File: config/src/main/java/com/typesafe/config/impl/ConfigNodeArray.java
package com.typesafe.config.impl;

import com.typesafe.config.ConfigOrigin;

import java.util.Collection;

final class ConfigNodeArray extends ConfigNodeComplexValue {
    ConfigNodeArray(Collection<AbstractConfigNode> children) {
        super(children);
    }

    ConfigNodeArray(Collection<AbstractConfigNode> children, ConfigOrigin origin) {
        super(children, origin);
    }

    @Override
    protected ConfigNodeArray newNode(Collection<AbstractConfigNode> nodes) {
        return new ConfigNodeArray(nodes, origin());
    }
}

File: config/src/main/scala/com/typesafe/config/impl/ConfigNodeArray.scala
package com.typesafe.config.impl

import java.{ util => ju }

final class ConfigNodeArray private[impl] (
    children: ju.Collection[AbstractConfigNode]) extends ConfigNodeComplexValue(children) {
    override def newNode(nodes: ju.Collection[AbstractConfigNode]) =
        new ConfigNodeArray(nodes)
}
